libretroshare/src/tcponudp/tcpstream.cc

```diff
@@ -2648,7 +2667,7 @@ int	setupBinaryCheck(std::string fname)
 /* uses seq number to track position -> ensure no rollover */
 int	checkData(uint8 *data, int size, int idx)
 {
-	if (bc_fd <= 0)
+	if (bc_fd == 0)
```

Comment:
despite the misleading name `bc_fd` is a pointer so check it as a pointer and avoid aritmetic

`if(!bc_fd)`

Reply:
Done.
But new commits was waiting so they are added before you req.